src/codeu/chat/client/core/Context.java

```diff
@@ -61,4 +68,39 @@ public ServerInfo getInfo() {
     return view.getInfo();
   }
 
+  public void deleteUser(User user) {
+    controller.deleteUser(user);
+  }
+
+  public void deleteConversation(ConversationHeader c) {
+    controller.deleteConversation(c);
+  }
+
+  public View getView() {
+    return (View)view;
+  }
+
+  public HashSet<Uuid> getNewAdmins() {
+    return (HashSet<Uuid>) view.getNewAdmins();
+  }
+
+  public HashSet<Uuid> getAdmins() {
+    return (HashSet<Uuid>) view.getAdmins();
+  }
+
+  public String getAuthInfo(Uuid id) {
+    return view.getAuthInfo(id);
+  }
+
+  public void writeAuthInfo(Uuid id, String password) {
+    controller.writeAuthInfo(id, password);
+  }
+
+  public void addAdmin(String name, boolean log) {
```

Comment:
Very good! This one perfectly done from client to context to controller to server. Just one thing you should add. Return a boolean from the server indicating whether adding the client was successful or not. Have the context return it all the way back to the client. If false, show a message to the user "Couldn't add admin". It's good for the user to know what happened after they tried to add an admin.